integrations/jira/jira/client.py

```diff
@@ -103,107 +179,105 @@ async def create_events_webhook(self, app_host: str) -> None:
             params["jql"] = config.selector.jql
             logger.info(f"Found JQL filter: {config.selector.jql}")
 
-        total_issues = (await self._get_paginated_issues(params))["total"]
+        async for issues in self._get_paginated_data(
+            f"{self.api_url}/search", "issues", initial_params=params
+        ):
+            yield issues
 
-        if total_issues == 0:
-            logger.warning(
-                "Issue query returned 0 issues, did you provide the correct Jira API credentials and JQL query?"
-            )
-
-        params["maxResults"] = PAGE_SIZE
-        while params["startAt"] <= total_issues:
-            logger.info(f"Current query position: {params['startAt']}/{total_issues}")
-            issue_response_list = (await self._get_paginated_issues(params))["issues"]
-            yield issue_response_list
-            params["startAt"] += PAGE_SIZE
+    async def get_single_user(self, account_id: str) -> Dict[str, Any]:
+        return await self._send_api_request(
+            "GET", f"{self.api_url}/user", params={"accountId": account_id}
+        )
 
-    async def get_paginated_users(
-        self,
-    ) -> AsyncGenerator[list[dict[str, Any]], None]:
+    async def get_paginated_users(self) -> AsyncGenerator[List[Dict[str, Any]], None]:
         logger.info("Getting users from Jira")
-
-        params = self._generate_base_req_params()
-
-        total_users = len(await self._get_users_data(params))
-
-        if total_users == 0:
-            logger.warning(
-                "User query returned 0 users, did you provide the correct Jira API credentials?"
-            )
-
-        params["maxResults"] = PAGE_SIZE
-        while params["startAt"] < total_users:
-            logger.info(f"Current query position: {params['startAt']}/{total_users}")
-
-            user_response_list = await self._get_users_data(params)
-
-            if not user_response_list:
-                logger.warning(f"No users found at {params['startAt']}")
-                break
-
-            logger.info(
-                f"Retrieved users: {len(user_response_list)} "
-                f"(Position: {params['startAt']}/{total_users})"
-            )
-
-            yield user_response_list
-            params["startAt"] += PAGE_SIZE
-
-    async def get_single_user(self, account_id: str) -> dict[str, Any]:
-        user_response = await self.client.get(
-            f"{self.api_url}/user", params={"accountId": account_id}
-        )
-        user_response.raise_for_status()
-        return user_response.json()
+        async for users in self._get_paginated_data(f"{self.api_url}/users/search"):
+            yield users
+
+    async def get_paginated_teams(
+        self, org_id: str
+    ) -> AsyncGenerator[List[Dict[str, Any]], None]:
+        logger.info("Getting teams from Jira")
+
+        selector = cast(TeamResourceConfig, event.resource_config).selector
+        include_members = selector.include_members
+
```

Comment:
move this should be in the main.py

A new convention we're following: only the main.py should have access to the resource config

Reply:
Okay. Thanks